stackslib/Cargo.toml

```diff
@@ -0,0 +1,128 @@
+version = "0.0.1"
+authors = [ "Jude Nelson <jude@stacks.org>",
+            "Aaron Blankstein <aaron@blockstack.com>",
+            "Ludo Galabru <ludovic@blockstack.com>" ]
+license = "GPLv3"
+homepage = "https://github.com/blockstack/stacks-blockchain"
+repository = "https://github.com/blockstack/stacks-blockchain"
+description = "Reference implementation of the Stacks 2.0 Blockchain"
+keywords = [ "stacks", "stx", "bitcoin", "crypto", "blockstack", "decentralized", "dapps", "blockchain" ]
+readme = "README.md"
+resolver = "2"
+edition = "2021"
+rust-version = "1.61"
+
+[profile.release]
+debug = true
+
+[lib]
+name = "blockstack_lib"
+path = "src/lib.rs"
+
+[[bin]]
+name = "stacks-inspect"
+path = "src/main.rs"
+
+[[bin]]
+name = "clarity-cli"
+path = "src/clarity_cli_main.rs"
+
+[[bin]]
+name = "blockstack-cli"
+path = "src/blockstack_cli.rs"
+
+[dependencies]
+rand = "0.7.3"
+rand_chacha = "=0.2.2"
+serde = "1"
+serde_derive = "1"
+serde_stacker = "0.1"
+sha3 = "0.10.1"
+ripemd = "0.1.1"
+regex = "1"
+mio = "0.6"
+lazy_static = "1.4.0"
+url = "2.1.0"
+percent-encoding = "2.1.0"
+prometheus = { version = "0.9", optional = true }
+integer-sqrt = "0.1.3"
+slog = { version = "2.5.2", features = [ "max_level_trace" ] }
+slog-term = "2.6.0"
+slog-json = { version = "2.3.0", optional = true }
+chrono = "0.4.19"
+libc = "0.2.82"
+clarity = { path = "../clarity" }
+stacks-common = { path = "../stacks-common" }
+pox-locking = { path = "../pox-locking" }
+siphasher = "0.3.7"
+
+[target.'cfg(unix)'.dependencies]
+nix = "0.23"
+
+[target.'cfg(windows)'.dependencies]
+winapi = { version = "0.3", features = ["consoleapi", "handleapi", "synchapi", "winbase"] }
+
+[target.'cfg(windows)'.dev-dependencies]
+winapi = { version = "0.3", features = ["fileapi", "processenv", "winnt"] }
+
+[dependencies.serde_json]
+version = "1.0"
+features = ["arbitrary_precision", "unbounded_depth"]
+
+[dependencies.secp256k1]
+version = "0.24.3"
+features = ["serde", "recovery"]
+
+[dependencies.rusqlite]
+version = "=0.24.2"
+features = ["blob", "serde_json", "i128_blob", "bundled", "trace"]
+
+[dependencies.ed25519-dalek]
+version = "=1.0.0-pre.3"
+features = ["serde"]
+
+[dependencies.curve25519-dalek]
+version = "=2.0.0"
+features = ["serde"]
+
+[dependencies.time]
+version = "0.2.23"
+features = ["std"]
+
+[dev-dependencies]
+assert-json-diff = "1.0.0"
+criterion = "0.3.5"
+stdext = "0.3.1"
+stx-genesis = { path = "../stx-genesis"}
+clarity = { features = ["default", "testing"], path = "../clarity" }
+stacks-common = { features = ["default", "testing"], path = "../stacks-common" }
+rstest = "0.17.0"
+rstest_reuse = "0.5.0"
+
+[features]
+default = ["developer-mode"]
+profile-sqlite = []
+disable-costs = []
+developer-mode = []
+monitoring_prom = ["prometheus"]
+slog_json = ["slog-json", "stacks-common/slog_json", "clarity/slog_json", "pox-locking/slog_json"]
+testing = []
+
+# Use a bit more than default optimization for
+#  dev builds to speed up test execution
```

Comment:
Remove this.  If anything, we want build times to be fast when developing.

Reply:
Can we make it in the next PR, I would like to make sure that it would be granular changes and easy to revert if something happen.

Reply:
This is present in the current `Cargo.toml` and should IMO be replicated here as @sergey-shandar did.

Opt level 1 has pretty small impacts on compile times but pretty big impacts on test execution. This and the setting below it yielded shaved like 25%-30% of the execution time off of unit tests when it was added to the toml file. Build times, on the other hand, are generally pretty quick if you don't need to rebuild dependencies (as would be the case for local development) and can take advantage of caching on github actions. Not to mention that build times for local developers should also improve dramatically as more pieces get refactored into other workspace members so that you need to rebuild less of the project to test any changes.

Reply:
Okay, fine to leave it.

Reply:
> This is present in the current `Cargo.toml` and should IMO be replicated here as @sergey-shandar did.
> 
> Opt level 1 has pretty small impacts on compile times but pretty big impacts on test execution. This and the setting below it shaved like 25%-30% of the execution time off of unit tests when it was added to the toml file. Build times, on the other hand, are generally pretty quick if you don't need to rebuild dependencies (as would be the case for local development) and can take advantage of caching on github actions. Not to mention that build times for local developers should also improve dramatically as more pieces get refactored into other workspace members so that you need to rebuild less of the project to test any changes.

IMHO, we should do all optimization it on workspace level.